zbus/src/connection/socket/unix.rs

```diff
@@ -359,11 +364,21 @@ fn get_unix_peer_creds_blocking(fd: RawFd) -> io::Result<crate::fdo::ConnectionC
             .unwrap_or(Vec::new());
         groups.sort_by_key(|gid| gid.as_raw());
 
+        let pidfd = match getsockopt(&fd, PeerPidfd) {
+            Err(Errno::ENOPROTOOPT) => None,
+            Ok(pidfd) => Some(OwnedFd::from(pidfd)),
+            Err(e) => return Err(e.into()),
+        };
+
         creds = creds.set_unix_user_id(uid.as_raw()).set_process_id(pid);
 
         for group in groups.iter() {
             creds = creds.add_unix_group_id(group.as_raw());
         }
+
+        if let Some(pidfd) = pidfd {
```

Comment:
same comment here as for groups: best to just do this directly (in the `match` arm above).

Reply:
Had that before, but then some state was set in there, other was required later in the code, so I split it into fetching the state and applying it and it seemed better to me. I can do it if you feel strongly about it.

Reply:
If there is no reason to do it in two phases, I think we shouldn't. The less logic and state, the easier to read. Although I think it should be all done here since pidfd isn't more important than uid IMO.

Reply:
Changed it to mutate creds as early as possible.